### PR TITLE
Fix: Switch component not being styled as disabled when is checked

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**What is this feature?**

This PR fixes Switch component not having any visual indication that the Switch is checked when is also disabled.

**Special notes for your reviewer:**

Before the change:

Screenshot 2024-01-04 at 13 09 44

After the change:
Screenshot 2024-01-04 at 13 04 44

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/Switch/Switch.tsx [hunk 1]
```diff
@@ -104,6 +99,24 @@ const getSwitchStyles = (theme: GrafanaTheme2, transparent?: boolean) => ({
...
+        cursor: 'not-allowed',
+
+        '&:hover': {
+          background: theme.colors.action.disabledBackground,
+        },
+      },
+
+      '&:disabled:checked + label': {
+        '&::after': {
+          transform: 'translate3d(18px, -50%, 0)',
```
Comment (JoaoSilvaGrafana) [inline packages/grafana-ui/src/components/Switch/Switch.tsx:115]: Don't think this is necessary for this case right? It should already be set in line 97

File: packages/grafana-ui/src/components/Switch/Switch.tsx [hunk 2]
```diff
@@ -104,6 +99,24 @@ const getSwitchStyles = (theme: GrafanaTheme2, transparent?: boolean) => ({
         },
       },
 
+      '&:disabled + label': {
+        background: theme.colors.action.disabledBackground,
+        borderColor: theme.colors.border.weak,
+        color: theme.colors.text.disabled,
```
Comment (JoaoSilvaGrafana) [inline packages/grafana-ui/src/components/Switch/Switch.tsx:105]: Is the change in color here needed? 🤔

File: packages/grafana-ui/src/components/Switch/Switch.tsx (hunk 1 above)
Comment (soniaAguilarPeiron) [reply packages/grafana-ui/src/components/Switch/Switch.tsx:115]: you're right!

File: packages/grafana-ui/src/components/Switch/Switch.tsx (hunk 2 above)
Comment (soniaAguilarPeiron) [reply packages/grafana-ui/src/components/Switch/Switch.tsx:105]: good catch! removing it!

**Review by JoaoSilvaGrafana — APPROVED** [review]
Nice, thanks for this!

**Pull request merged**